packages/strapi-bookshelf/lib/index.js

```diff
@@ -482,6 +482,194 @@ module.exports = function(strapi) {
+              }
+
+              break;
+            case 'oneToMany':
+            case 'manyToOne':
+            case 'manyToMany':
+              if (details.dominant === true) {
+                acc[current] = params.values[current];
+              } else if (response[current] && _.isArray(response[current]) && current !== 'id') {
+                // Records to add in the relation.
+                const toAdd = _.differenceWith(params.values[current], response[current], (a, b) =>
+                  ((typeof a === 'number') ? a : a[Model.primaryKey].toString()) === b[Model.primaryKey].toString()
+                );
+                // Records to remove in the relation.
+                const toRemove = _.differenceWith(response[current], params.values[current], (a, b) =>
+                  a[Model.primaryKey].toString() === ((typeof b === 'number') ? b : b[Model.primaryKey].toString())
+                )
+                  .filter(x => toAdd.find(y => x.id === y.id) === undefined);
+
+                // Push the work into the flow process.
+                toAdd.forEach(value => {
+                  value = (typeof value === 'number') ? { id: value } : value;
+
+                  value[details.via] = parseFloat(params[Model.primaryKey]);
+                  params.values[Model.primaryKey] = parseFloat(params[Model.primaryKey]);
+
+                  virtualFields.push(this.addRelation(models, models[details.model || details.collection], {
+                    id: value[Model.primaryKey] || value.id || value._id,
+                    values: association.nature === 'manyToMany' ? params.values : value,
+                    foreignKey: current
+                  }));
+                });
+
+                toRemove.forEach(value => {
+                  value[details.via] = null;
+
+                  virtualFields.push(this.removeRelation(models, models[details.model || details.collection], {
+                    id: value[Model.primaryKey] || value.id || value._id,
+                    values: association.nature === 'manyToMany' ? params.values : value,
+                    foreignKey: current
+                  }));
+                });
+              } else if (_.get(Model._attributes, `${current}.isVirtual`) !== true) {
+                acc[current] = params.values[current];
+              }
+
+              break;
+            default:
+          }
+        }
+
+        return acc;
+      }, {});
+
+      if (!_.isEmpty(values)) {
+        virtualFields.push(Model
+          .forge({
+            [Model.primaryKey]: params[Model.primaryKey]
+          })
+          .save(values, {
+            patch: true
+          }));
+      } else {
+        virtualFields.push(Promise.resolve(_.assign(response, params.values)));
+      }
+
+      // Update virtuals fields.
+      const process = await Promise.all(virtualFields);
```

Comment:
`return await Promise.all(virtualFields);`?

You are returning nothing, this value could be useful for the user.